### ISSUE TITLE
Source Code Not Recognized in PDF Files in Version 0.0.6

### ISSUE THREAD

**Issue by yewool0818** [pymupdf/RAG#61]
Hello pymupdf4llm Maintainers,

I've recently upgraded to version 0.0.6 of the pymupdf4llm library and encountered an issue where the library fails to recognize source code within PDF files. In the previous version (0.0.5), the source code in PDFs was recognized and processed without any issues.

##### Environment
- windows 
- python 3.12 

##### Steps to Reproduce
1. Upgrade pymupdf4llm to version 0.0.6.
2. Load a PDF file containing source code.
3. Attempt to process the file using the library's relevant functions.

##### Expected Behavior
The library should recognize and extract the source code from the PDF.

##### Actual Behavior
The library does not recognize the source code in the PDF.

##### Additional Information
The issue was not present in version 0.0.5.
No changes were made to the PDF files between tests.

I would appreciate any guidance on this issue or any updates regarding a potential fix.
Thank you for your attention to this matter.

**Comment by JorjMcKie** [comment]
Please provide the example PDF. Otherwise there is no way to deal with the problem.

**Comment by yewool0818** [comment]
@JorjMcKie 

[sample_pdf_data.pdf](https://github.com/user-attachments/files/16006913/sample_pdf_data.pdf)
This is the sample pdf file.

It looks like that
![image](https://github.com/pymupdf/RAG/assets/65100583/ed7f981a-844e-42ef-83d4-a980fcb8f018)

but after converting to markdown, code line is disappeared.
```markdown
# convert PDF to Markdown Test

This is my code.


result


but code is not recognized.
```

I used this code to converting.
```python
def pdf_to_markdown(pdf_file, t_md_file):
    md_text = pymupdf4llm.to_markdown(pdf_file)
    print(md_text)
    # Split the text into lines
    lines = md_text.split('\n')

    # Patterns to match entire line
    patterns = [r"^\*\*(\d+) ([^\*]+)\*\* \*\*(\d+)\*\*$", r"^-----$"]

    # Filter lines
    filtered_lines = [line for line in lines if not any(re.match(pattern, line) for pattern in patterns)]

    # Join the filtered lines back into a single string
    filtered_text = '\n'.join(filtered_lines)
    with open(t_md_file, 'w', encoding='utf-8') as f:
        f.write(filtered_text + '\n')
```

**Comment by JorjMcKie** [comment]
Thank you for the file. This is an interesting situation.

The new version has improved logic to differentiate between vector graphics that simply is background shading for text (thus insignificant) and something potentially significant like Gantt charts.

This is done by checking whether only "fill" graphics are present. If there are "stroke" type graphics, then this is assumed to be potentially significant drawings.

Your example contains shaded boxes that also have stroked lines (the borders around your code pieces) with rounded corners.
Because of the borders, there is no way to determine that these drawings are insignificant and can really be ignored.

If you remove the borders of the shaded code pieces, things will work.

What also will work is this:
```python
import pathlib

import pymupdf4llm

print(f"{pymupdf4llm.version=}")
md = pymupdf4llm.to_markdown("test.pdf", write_images=True)
pathlib.Path("test.md").write_bytes(md.encode())
```
The resulting markdown looks like this:
```text
# convert PDF to Markdown Test

This is my code.

![test.pdf-0-0.png](test.pdf-0-0.png)

result

![test.pdf-0-1.png](test.pdf-0-1.png)

but code is not recognized.


-----
```
In other words: The code pieces are interpreted as significant graphics that must be converted to images. When rendering the markdown file, it will look like this:
 
![image](https://github.com/pymupdf/RAG/assets/8290722/8577cc4b-df1f-4fca-9e54-8668ec05b016)

**Comment by pjawale-cohere** [comment]
@JorjMcKie Is there no way to install the older version? The new update seems to be breaking my extractions scripts as well. 
Thanks!
